src/Client.php

```diff
@@ -173,87 +179,129 @@ public function userBalance(string $report_type_uid, bool $detailed = false): Us
+                $query['_query'] = $filter_query;
+            }
+
+            if (\is_int($per_page = $params->getPerPage())) {
+                $query['_size'] = \max(1, $per_page);
+            }
+
+            if (\is_int($offset = $params->getOffset())) {
+                $query['_offset'] = $offset;
+            }
+
+            if (\is_int($page = $params->getPage())) {
+                $query['_page'] = \max(1, $page);
+            }
+
+            if (\is_string($sort_by = $params->getSortBy())) {
+                $query['_sort'] = $sort_by;
+            }
+
+            if (\is_bool($is_calc_total = $params->isCalcTotal())) {
+                $query['_calc_total'] = $is_calc_total ? 'true' : 'false';
+            }
+
+            if (\is_bool($is_detailed = $params->isDetailed())) {
+                $query['_detailed'] = $is_detailed ? 'true' : 'false';
+            }
+        }
+
         return UserReportsResponse::fromHttpResponse(
-            $this->doRequest(new Request('get', 'user/reports'), [
-                'query' => [
-                    '_content'    => $content
-                        ? 'true'
-                        : 'false',
-                    '_query'      => $query,
-                    '_size'       => \max(1, $size),
-                    '_offset'     => \max(0, $offset),
-                    '_page'       => \max(1, $page),
-                    '_sort'       => $sort,
-                    '_calc_total' => $calc_total
-                        ? 'true'
-                        : 'false',
-                    '_detailed'   => $detailed
-                        ? 'true'
-                        : 'false',
-                ],
-            ])
+            $this->doRequest(new Request('get', 'user/reports'), ['query' => $query])
         );
     }
 
     /**
      * {@inheritdoc}
      */
-    public function userReport(string $report_uid,
-                               bool $content = true,
-                               bool $detailed = true): UserReportResponse
+    public function userReport(UserReportParams $params): UserReportResponse
     {
         return UserReportResponse::fromHttpResponse(
-            $this->doRequest(new Request('get', \sprintf('user/reports/%s', \urlencode($report_uid))), [
+            $this->doRequest(new Request('get', \sprintf('user/reports/%s', \urlencode($params->getReportUid()))), [
                 'query' => [
-                    '_content'  => $content
-                        ? 'true'
-                        : 'false',
-                    '_detailed' => $detailed
-                        ? 'true'
-                        : 'false',
+                    '_content'  => $params->isIncludeContent() === false
+                        ? 'false'
+                        : 'true',
+                    '_detailed' => $params->isDetailed() === false
+                        ? 'false'
+                        : 'true',
```

Comment:
```suggestion
                    '_detailed' => $params->isDetailed() === true
                        ? 'true'
                        : 'false',
```

Reply:
Here's the same logic as in the [comment above](https://github.com/avtocod/b2b-api-php/pull/22#discussion_r509342841).